src/utils/helpers/GraphHelpers.ts

```diff
@@ -1,187 +1,298 @@
+    const svgContainerNode = select<SVGSVGElement, DependencyNode>('#container').node();
+    const dim = findGroupBackgroundDimension(highlightedNodes.data());
+
+    if (!svgContainerNode || !dim) {
+        return;
+    }
+
+    const width = Number(svgContainerNode.getAttribute('width'));
+    const height = Number(svgContainerNode.getAttribute('height'));
+
+    const scaleValue = Math.min(8, 0.9 / Math.max(dim.width / width, dim.height / height));
+
     svgContainer
-        .append('svg:defs')
-        .append('svg:marker')
-        .attr('id', 'provider')
-        .attr('viewBox', '-5 -5 40 10')
-        .attr('refX', 15)
-        .attr('refY', 0)
-        .attr('markerWidth', 40)
-        .attr('markerHeight', 40)
-        .attr('orient', 'auto')
-        .append('svg:path')
-        .attr('d', 'M0,-5L20,0L0,5,q10 -5,0 -10')
-        .attr('fill', '#E5E5E6');
-}
-
-export function generateLabelPath(this: Node, node: DependencyNode) {
-    const labelTextDimensions = getLabelTextDimensions(this);
-
-    if (!labelTextDimensions) {
-        return '';
+        .transition()
+        .duration(750)
+        .call(
+            zoom<any, any>().on('zoom', zoomed).transform,
+            zoomIdentity
+                .translate(width / 2, height / 2)
+                .scale(scaleValue)
+                .translate(-dim.x - dim.width / 2 - 250, -dim.y - dim.height / 2)
+        );
+}
+
+export function setDependencyLevelOnEachNode(clickedNode: DependencyNode, nodes: DependencyNode[]): DependencyNode[] {
+    nodes.forEach((node: DependencyNode) => (node.level = 0));
+
+    const visitedNodes: DependencyNode[] = [];
+    const nodesToVisit: DependencyNode[] = [];
+
+    nodesToVisit.push({ ...clickedNode, level: 1 });
+
+    while (nodesToVisit.length > 0) {
+        const currentNode = nodesToVisit.shift();
+
+        if (!currentNode) {
+            return [];
+        }
+
+        currentNode.links.forEach((node: DependencyNode) => {
+            if (!contains(visitedNodes, node) && !contains(nodesToVisit, node)) {
+                node.level = currentNode.level + 1;
+                nodesToVisit.push({ ...node, level: currentNode.level + 1 });
+            }
+        });
+
+        visitedNodes.push(currentNode);
     }
 
-    const labelTextWidth = labelTextDimensions.width;
+    return visitedNodes;
+}
+
+function contains(arr: DependencyNode[], node: DependencyNode) {
+    return arr.findIndex((el: DependencyNode) => Boolean(el.name === node.name && el.version === node.version)) > -1;
+}
 
+export function areNodesConnected(a: DependencyNode, b: DependencyNode, links: DependencyLink[]) {
+    return (
+        a.index === b.index ||
+        links.some(
+            link =>
+                (link.source.index === a.index && link.target.index === b.index) ||
+                (link.source.index === b.index && link.target.index === a.index)
+        )
+    );
+}
+
+export function getHighLightedLabelColor(node: DependencyNode) {
```

Comment:
typo ?
``getHighLightedLabelColor`` instead ``getHighlightedLabelColor`` ?

Reply:
I don't know why but it looked beter with two words. Gonna change.